src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -396,6 +384,19 @@ public void setBaitSetName(final String name) {
 
         /** Adds information about an individual SAMRecord to the statistics. */
         public void acceptRecord(final SAMRecord record) {
+            // Just ignore secondary alignments altogether
+            if (record.getNotPrimaryAlignmentFlag()) return;
+
+            // Cache some things, and compute the total number of bases aligned in the record.
+            final boolean mappedInPair = record.getReadPairedFlag() && !record.getMateUnmappedFlag() && !record.getSupplementaryAlignmentFlag();
```

Comment:
we use it after returning for an unmapped read (and duplicate) but it would be good to either document that assumption or move it after those checks.